solidity/contracts/middleware/InterchainQueryRouter.sol

```diff
@@ -70,25 +73,34 @@ contract InterchainQueryRouter is Router, IInterchainQueryRouter {
         );
     }
 
+    function query(
```

Comment:
Ditto, I think four flavors of query is too many, to the point where the lack of clarity outweighs the optionality